### ISSUE TITLE
Field-level encrypted values

### ISSUE THREAD

**Issue by themightychris** [JarvusInnovations/gitsheets#143]
Allow specific fields on a sheet to be marked as encrypted-at-rest. The TOML record on disk carries ciphertext for those fields; the API decrypts on read.

## Motivation

Some use cases want gitsheets's single-repo simplicity AND public hosting AND private fields — typically: an internal-but-public-ish dashboard, a public ledger with redacted columns, an open-data project with optional PII alongside the public records.

Today's options for such a consumer are: split into a second private repo (loses the single-repo simplicity), or don't use gitsheets for those fields. Field-level encryption fills the gap.

## Sketch

Schema declares which fields are encrypted:

\`\`\`toml
[gitsheet.schema.properties.email]
type = "string"
encrypted = true
\`\`\`

On write, the library encrypts the field's value before serializing the record. On read, it decrypts. The record file carries ciphertext at that path; the rest of the record stays plaintext.

\`\`\`toml
slug = "janedoe"
email = "enc:v1:base64ciphertext..."
fullName = "Jane Doe"
\`\`\`

## Design decisions to make

- **Encryption mode** — AES-SIV (deterministic, byte-stable, leaks "same plaintext = same ciphertext") vs. AES-GCM with random IV (non-deterministic, breaks canonical normalization)
- **Salt / mixing** — a server-secret salt mixed into the encryption to defeat pre-computed dictionary attacks across public repos
- **Versioning** — \`enc:v1:...\` tags so keys can be rotated lazily
- **HMAC index field** — for searchable equality lookup on encrypted fields, store a separate \`<field>Hash\` (HMAC with a server-secret HMAC key) alongside the ciphertext
- **Size padding** — pad ciphertext to fixed-length buckets per field type so ciphertext length doesn't leak plaintext length
- **Key storage** — how the consumer provides the key(s) to the library (env var, KMS, callback function)
- **Scrubbed snapshot integration** — \`git sheet scrub\` (or equivalent) replaces encrypted fields with \`[REDACTED]\` literals so the snapshot drops the ciphertexts entirely

## Known limits

- **Permanent breach risk** — once published to a public repo, ciphertexts are harvested forever; key compromise at any point in time exposes all historical encrypted values
- **Legal gray area** — encrypted PII in a public location may still constitute regulated processing under GDPR/CCPA-style regimes
- **Key loss = data loss** — losing the encryption key permanently loses access to the encrypted fields
- **Implementation surface** — encryption module, key-loading API, search-via-HMAC layer, key-rotation tooling, scrubbing integration

These limits make field-level encryption a *fit-for-purpose tool, not a default*. Consumers should use it only when public-repo visibility + selective opacity is genuinely the right trade.

## Status

Backlog. Surfaced from the codeforphilly.org rewrite design discussion — that consumer ended up choosing a separate-private-bucket approach because their PII surface is small and the forever-breach risk of public ciphertext felt worse than the operational cost of a separate private store. Field-level encryption remains a real option for *other* consumers; filing here so the work is tracked when there's demand.

## Coordinates with

- behaviors/validation.md / behaviors/normalization.md — the encryption layer runs adjacent to validation and normalization in the write pipeline
- api/sheet.md — Sheet.upsert / query / findByIndex transparently encrypt/decrypt
- #136 error taxonomy — new error class? EncryptionError, or fold into ConfigError?
